src/main/java/nwoolcan/controller/brewery/BreweryControllerImpl.java

```diff
@@ -20,8 +37,71 @@ public BreweryControllerImpl(final Brewery brewery) {
         this.warehouseController = new WarehouseControllerImpl(brewery.getWarehouse());
     }
 
+    @Override
+    public Collection<MasterBatchViewModel> getBatches(final QueryBatch queryBatch) {
+        return brewery.getBatches(queryBatch)
+                      .stream()
+                      .map(MasterBatchViewModel::new)
+                      .collect(Collectors.toList());
+    }
+
+    @Override
+    public Result<Empty> stockBatch(final int batchId, final int beerArticleId, final Date expirationDate) {
+        final Result<Batch> batchResult = getBatchById(batchId);
+        final Result<BeerArticle> beerArticleResult = getBeerArticleById(beerArticleId);
+        return Result.ofEmpty()
```

Comment:
```suggestion
                     return getBatchById(batchId).flatMap(...
                     
```
starting with Result.ofEmpty() is useless since you map to batchResult right after. Make more sense to start with batchResult then